src/notation/internal/notationinteraction.cpp

```diff
@@ -764,8 +865,44 @@ void NotationInteraction::doSelect(const std::vector<EngravingItem*>& elements,
             score()->setUpdateAll();
         }
 
-        if (elements.front()->selected()) {
-            score()->deselect(elements.front());
+        std::vector<EngravingItem*> ll = elementsNear(m_logicClickPos);
```

Comment:
I feel like this code doesn't really belong inside this method, but rather somewhere in NotationViewInputController.

`NotationInteraction::doSelect` is a relatively "low-level" method, performing just the selection process itself. The new code is much more "high-level", since it's about interaction logic and UX. Stuff like that fits better in `NotationViewInputController`. 

Moving this code may mean that you have to expose some methods of `NotationInteraction` to the outside world; for example, you may need to add a `deselect` method (although `clearSelection` will probably also do the job), and ~`elementsNear`~ `hitElements` may need to be made public. Retrieving the current selection can be done using `viewInteraction()->selection()->…` (see `INotationSelection`).

Reply:
Yeah, I will move it to `NotationViewInputController`